### PR TITLE
[MDS-5791] File Upload Bug fix and some improvements

### PR DESCRIPTION
**Pull request by isuru-aot**
## Objective 

[MDS-5791](https://bcmines.atlassian.net/browse/MDS-5791)

_Why are you making this change? Provide a short explanation and/or screenshots_
Bug is fixed with some other improvements.

### CODE REVIEW

File: services/minespace-web/src/components/pages/Project/ProjectSummaryPage.tsx [hunk 1]
```diff
@@ -272,17 +272,20 @@ export const ProjectSummaryPage: FC<ProjectSummaryPageProps> = (props) => {
...
-        filtered
-      );
+    if (values.authorizations !== undefined && values.authorizations !== null) {
+      for (const code of Object.keys(values.authorizations)) {
+        if (values.authorizations?.[code] !== undefined) {
+          const file_to_upload =
+            values.authorizations?.[code]?.AMENDMENT[0]?.amendment_documents ?? [];
+          const filtered = file_to_upload.filter((doc) => !doc.mine_document_guid);
+          values.authorizations[code].AMENDMENT[0].amendment_documents = filtered;
+          change(
```
Comment (henryoforeh-dev) [inline services/minespace-web/src/components/pages/Project/ProjectSummaryPage.tsx]: Dispatch also needs to happen here. Code on line 281 is not required.

File: services/common/src/components/projectSummary/AuthorizationsInvolved.tsx [hunk 2]
```diff
@@ -177,7 +182,8 @@ const RenderEMANewPermitSection = ({
   project_summary_guid,
 }) => {
   const new_props = {
-    ...props.formValues.authorizations.AIR_EMISSIONS_DISCHARGE_PERMIT.AMENDMENT[0],
+    ...props.formValues.authorizations[code],
```
Comment (matbusby-fw) [inline services/common/src/components/projectSummary/AuthorizationsInvolved.tsx]: Should this change include the `AMENDMENT[0]` as it did before?

```suggestion
    ...props.formValues.authorizations[code].AMENDMENT[0],
```

**Review by matbusby-fw — CHANGES_REQUESTED** [review]
I think it's looking mostly good.  Just what Henry mentioned and one small change to keep it inline with the previous functionality.

**Comment by isuru-aot** [comment]
Removed the `dispatch(change(....` and moved the logic to BE.

File: services/core-api/app/api/projects/project_summary/models/project_summary.py [hunk 3]
```diff
@@ -296,16 +296,19 @@ def create_or_update_authorization(self, authorization):
...
-                            document_manager_guid=doc.get('document_manager_guid'))
-                        project_summary_authorization_doc = ProjectSummaryAuthorizationDocumentXref(
-                            mine_document_guid=mine_doc.mine_document_guid,
-                            project_summary_authorization_guid=updated_authorization.project_summary_authorization_guid,
-                            project_summary_document_type_code=doc.get('project_summary_document_type_code'))
-                        project_summary_authorization_doc.mine_document = mine_doc
-                        updated_authorization.amendment_documents.append(project_summary_authorization_doc)
+                        if doc.get('mine_document_guid') is None:
+                            print("----mine_document_guid: ", doc.get('document_name'))
+                            print("----mine_document_guid: ", doc.get('mine_document_guid'))
```
Comment (henryoforeh-dev) [inline services/core-api/app/api/projects/project_summary/models/project_summary.py]: Do you still need the print statements here ?

**Review by asinn134 — APPROVED** [review]

**Review by henryoforeh-dev — APPROVED** [review]

**Review by matbusby-fw — APPROVED** [review]

**Pull request merged**